SCAutolib/models/user.py

```diff
@@ -8,52 +8,116 @@
 The classes implement add_user and delete_user methods which can be used to
 create or remove a specified user in the system or in the specified IPA server.
 """
-from pathlib import Path
+from shutil import rmtree
 
+import json
+import pwd
+import python_freeipa
+from pathlib import Path, PosixPath
+
+from SCAutolib import run, logger, LIB_DUMP_USERS, LIB_DUMP_CARDS
+from SCAutolib.exceptions import SCAutolibException
+from SCAutolib.models import card as card_model
 from SCAutolib.models.CA import IPAServerCA
-from SCAutolib import run, logger
+from SCAutolib.models.file import OpensslCnf
+
+
+class BaseUser:
+    username: str = None
```

Comment:
Defining class attributes is more like good practice in Python. The reasons for this are simple:
- better readability
- better transparency of what is in the class
- no avoid collisions in instance attribute name and class method
- minor advantage also is that this helps editor to provide better code hints when writing the code

Reply:
Exactly as you said, it is good to be explicit in Python, so that is the summary of why to have class attributes.

Regarding the sources for this good practice, I don't have it because this is what I saw at different conferences and articles about Python development.